inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/scheduling/tasks/PredictionTask.java

```diff
@@ -62,66 +73,88 @@ public void run()
     {
         User user = getUser();
 
-        Predictions model = new Predictions(getProject(), getUser()); 
+        Project project = getProject();
+        Predictions model = new Predictions(project, getUser());
+        List<AnnotationDocument> documents = documentService.listAnnotationDocuments(project, user);
 
-        for (AnnotationLayer layer : annoService.listAnnotationLayer(getProject())) {
-            if (!layer.isEnabled()) {
+        for (AnnotationDocument document : documents) {
+            JCas jCas;
+            try {
+                jCas = documentService.readAnnotationCas(document);
```

Comment:
The CAS we obtain at this point may not have the correct type system. E.g. if I open a legacy project that was created before the introduction of the `PredictedSpan`, then it does not contain the type definition for that type and the prediction task fails. Hence, it is necessary to "upgrade" the CAS here to the latest type system by calling `AnnotationSchemaServiceImpl.upgradeCas(CAS, SourceDocument, String)`.

The same probably goes for the evaluation task.